From 54028ca373a2aca8980f35732cecc1bf4d11cbca Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Wed, 18 Mar 2015 14:44:42 +0800
Subject: [PATCH 3/3] atomisp: min freq lower to 100MHz on cht

ISP should lower its freq to 100MHz during idle status on cht.

Change-Id: I3ef082f77d9ba1c22e8258cd4f58ec10142f0064
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26573
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/47015
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp-regs.h     | 1 +
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_dfs_tables.h      | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp-regs.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp-regs.h
index 2ea28ce..e96818b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp-regs.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp-regs.h
@@ -148,6 +148,7 @@
 #define ISP_FREQ_320MHZ				0x140
 #define ISP_FREQ_266MHZ				0x10a
 #define ISP_FREQ_200MHZ				0xc8
+#define ISP_FREQ_100MHZ				0x64
 
 #define HPLL_FREQ_800MHZ			0x320
 #define HPLL_FREQ_1600MHZ			0x640
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_dfs_tables.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_dfs_tables.h
index 1307696..714b510 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_dfs_tables.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_dfs_tables.h
@@ -345,7 +345,7 @@ static const struct atomisp_freq_scaling_rule dfs_rules_cht[] = {
 };
 
 static const struct atomisp_dfs_config dfs_config_cht = {
-	.lowest_freq = ISP_FREQ_200MHZ,
+	.lowest_freq = ISP_FREQ_100MHZ,
 	.max_freq_at_vmin = ISP_FREQ_356MHZ,
 	.highest_freq = ISP_FREQ_356MHZ,
 	.dfs_table = dfs_rules_cht,
-- 
1.9.1

